compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3045,37 +3045,6 @@ generateS390CompareAndBranchOpsHelper(TR::Node * node, TR::CodeGenerator * cg, T
       // yay! success.. nothing else to do
       }
 
-   // ificmpyy
-   //   c2i        ; refcnt-1, unevaluated
-   //     icload   ; evaluated
-   //   c2i        ; refcnt-1, unevaluated
-   //     icload   ; evaluated
-   //
-   // Try to generate a CR.
-   // FIXME: can't the binary commutative analyser handle this? that's where this should be done
-   //            what about su2i? s2i? mixing of these? bytes? ints? longs?
-   //
-   else if (firstChild->getOpCodeValue()==TR::su2i && firstChild->getRegister()==NULL && firstChild->getReferenceCount()==1 &&
```

Comment:
I'm not seeing an s2i partner for this so what is the logic for eliminating this all together rather than converting it to s type?

Reply:
Fixed in d89a0b3